src/commcare_cloud/commands/deploy/formplayer.py

```diff
@@ -35,7 +36,12 @@ def deploy_formplayer(environment, args):
     print(color_notice("\nPreparing to deploy Formplayer to: "), end="")
     print(f"{environment.name}\n")
 
-    diff = get_deploy_diff(environment)
+    repo = None
+    if github_auth_provided():
```

Comment:
Auth is used for two things, I believe:
1. To obtain write permissions to set a deploy tag in the repo
2. To avoid being rate limited by github when making a bunch of requests for PR info.

Auth is not needed for other github operations, like fetching the latest commit.

We might consider separating these concerns, or at least being more explicit about what we require when, since this will likely be run by users who don't need or want github auth, those who do want auth for full deploy diffs, and Dimagi users who also have write access to create tags.  I'm not certain we handle all those situations correctly, and it seems prone to confusion.